code/controllers/subsystem/job.dm

```diff
@@ -590,14 +603,190 @@ SUBSYSTEM_DEF(job)
+/datum/controller/subsystem/job/proc/regenerate_job_config(mob/user)
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
+	var/list/file_data = list()
+
+	if(!fexists(file(toml_file))) // You need an existing (valid) TOML for this to work. Sanity check if someone calls this directly instead of through 'Generate Job Configuration' verb.
+		to_chat(user, span_notice("No jobconfig.toml found in the config folder! If this is not expected, please notify a server operator or coders. You may need to generate a new config file by running 'Generate Job Configuration' from the Server tab."))
+		return FALSE
+
+	var/job_config = rustg_read_toml_file(toml_file)
+	for(var/datum/job/occupation as anything in joinable_occupations)
+		var/job_name = occupation.title
+		var/job_key = occupation.config_tag
+
+		var/default_positions = job_config[job_key]["Total Positions"]
+		var/starting_positions = job_config[job_key]["Spawn Positions"]
+		var/playtime_requirements = job_config[job_key]["Playtime Requirements"]
+		var/required_account_age = job_config[job_key]["Required Account Age"]
+
+		// When we regenerate, we want to make sure commented stuff stays commented, but we also want to migrate information that remains uncommented. So, let's make sure we keep that pattern.
+		if(job_config["[job_key]"]) // Let's see if any data for this job exists.
+			if(file_data["[job_key]"]) // Sanity, let's just make sure we don't overwrite anything or add any dupe keys.
+				continue
+			if(default_positions) // If the variable exists, we want to ensure it migrated into the new TOML uncommented, to allow for flush migration.
+				file_data["[job_key]"] += list(
+					"Total Positions" = default_positions,
+				)
+			else // If we can't find anything for this variable, then we just throw in the codebase default with it commented out.
+				file_data["[job_key]"] += list(
+					"#Total Positions" = occupation.total_positions,
+				)
+
+			if(starting_positions) // Same pattern as above.
+				file_data["[job_key]"] += list(
+					"Spawn Positions" = starting_positions,
+				)
+			else
+				file_data["[job_key]"] += list(
+					"#Spawn Positions" = occupation.spawn_positions,
+				)
+
+			if(playtime_requirements) // Same pattern as above.
+				file_data["[job_key]"] += list(
+					"Playtime Requirements" = playtime_requirements,
+				)
+			else
+				file_data["[job_key]"] += list(
+					"#Playtime Requirements" = occupation.exp_requirements,
+				)
+
+			if(required_account_age) // Same pattern as above.
+				file_data["[job_key]"] += list(
+					"Required Account Age" = required_account_age,
+				)
+			else
+				file_data["[job_key]"] += list(
+					"#Required Account Age" = occupation.minimal_player_age,
+				)
+			continue
+		else
+			to_chat(user, span_notice("New job [job_name] (using key [job_key]) detected! Adding to jobconfig.toml using default codebase values..."))
```

Comment:
I thought MSO didn't want you to use default codebase values (and I tend to agree), what was the deal with that?

Reply:
no i wanted them to, i just didn't want it to get perma saved into the generated config

Reply:
Oh I see, they're commented out, okay that makes more sense to me